lib/signature.js

```diff
@@ -0,0 +1,163 @@
+var crypto = require('crypto');
+var querystring = require('querystring');
+var sortKeys = require('sort-object');
+var Buffer = require('safe-buffer').Buffer;
+var scmp = require('scmp');
+var extend = Object.assign ? Object.assign : require('util')._extend;
+
+var MESSAGEBIRD_REQUEST_TIMESTAMP = 'messagebird-request-timestamp';
+var MESSAGEBIRD_REQUEST_SIGNATURE = 'messagebird-signature';
+var MESSAGEBIRD_REQUEST_TIMEOUT = 100;
+var MESSAGEBIRD_REQUEST_HASH = 'sha256';
+
+/**
+ * Returns request is expired or not.
+ *
+ * @param {Object} req
+ * @return {Boolean}
+ */
+function isRecent(req) {
+  var timestamp = req.headers[MESSAGEBIRD_REQUEST_TIMESTAMP];
+  var currentTime = Math.floor(new Date().getTime() / 1000);
+
+  if (!timestamp) {
+    throw new Error('The "MessageBird-Request-Timestamp" header is missing.');
+  }
+
+  return (currentTime - timestamp) < MESSAGEBIRD_REQUEST_TIMEOUT;
+}
+
+/**
+ * Returns signatures are equal or not.
+ *
+ * @param {Object} req
+ * @param {Buffer} generatedSignature
+ * @return {Boolean}
+ */
+function isValid(req, generatedSignature) {
+  var signature = req.headers[MESSAGEBIRD_REQUEST_SIGNATURE];
+
+  if (!signature) {
+    throw new Error('The "MessageBird-Signature" header is missing.');
+  }
+
+  return scmp(Buffer.from(signature, 'base64'), generatedSignature);
+}
+
+/**
+ * Returns object is empty or not.
+ *
+ * @param {Object} obj
+ * @return {Boolean}
+ */
+function isEmptyObject(obj) {
+  for(var key in obj) {
+    if (obj.hasOwnProperty(key)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+/**
+ * Returns empty string if body is object and has no key inside otherwise stringify the object.
+ *
+ * @param {Object} obj
+ * @return {Boolean}
+ */
+function stringifyBody(body) {
```

Comment:
Can `body` also be `null` for `204 no content` responses or is it always an empty string in those cases?